### ISSUE TITLE
AV1 support

### ISSUE THREAD

**Issue by RinCat** [mastodon/mastodon#19734]
### Steps to reproduce the problem

1. Upload a AV1 video file
2. Try to play the uploaded video

### Expected behaviour

The video file should be pass-through or convert to mp4

### Actual behaviour

The video become audio only

### Specifications

Mastodon v3.5.3

**Comment by leo60228** [comment]
Does your browser support AV1? Are you uploading it as .mp4 or .webm?

**Comment by RinCat** [comment]
> Does your browser support AV1? Are you uploading it as .mp4 or .webm?

Yes, my browser support it, and most browsers support it https://caniuse.com/?search=AV1

The video uploaded as webm. The host ffmpeg enabled AV1 support.

BTW, the file become mp3, not a black screen video.

**Comment by leo60228** [comment]
Okay, I can reproduce this with a .webm, but not with a .mp4 (which works fine).

**Comment by leo60228** [comment]
This seems to be due to [marcel](https://github.com/rails/marcel) not [recognizing codecs other than VP8/VP9](https://github.com/rails/marcel/blob/04de6218a271d0df5f0c7a72ab1b25814e935ad7/data/custom.xml#L77) as `video/webm`, leading to it falling back to `audio/webm`. This should possibly be reported there.

**Comment by RinCat** [comment]
OK, I will report to marcel.

**Comment by sambonbonne** [comment]
If anyone else cross this issue, there is a PR on Marcel to fix it: https://github.com/rails/marcel/pull/104

But no activity on Marcel repository for one year, we'll have to be patient.

**Comment by cosstab** [comment]
The PR was merged on September. I'm running a recently built version of Mastodon Glitch (v4.6.0-alpha.1+glitch).

I tried uploading an AV1 .webm but I get error "422 Validation failed: File has contents that are not what they are reported to be".

If I try with .mkv "422 Validation failed: File content type is invalid, File is invalid".

.mp4 works, but my 20MB AV1 video gets transcoded to a 99MB h264 video, and it takes a while to do so, which doesn't sound optimal...
